### PR TITLE
Avoid iterating results twice in `propagation.py`

### PR DESCRIPTION
**Pull request by Yash-10**
This tries to solve a "TODO" in the `propagation` module where the variable `results` was iterated twice. 

The resulting improvement is ~1s for propagation for 50 seconds.

Thanks!

---

~Re-triggering still shows the tests as canceled. Maybe some remote issue.~ Not now

### CODE REVIEW

**Review by jorgepiloto — APPROVED** [review]
Looks good to me @Yash-10, no more comments by my side. Let us wait for @astrojuanlu approval too 🚀

**Review by astrojuanlu — APPROVED** [review]
Yay, thanks @Yash-10 !

**Pull request merged**
